src/Core/OrganizationFeatures/OrganizationSponsorships/FamiliesForEnterprise/Cloud/CloudSyncSponsorshipsCommand.cs

```diff
@@ -0,0 +1,98 @@
+using System.Linq;
+
+using System.Collections;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Bit.Core.Entities;
+using Bit.Core.Exceptions;
+using Bit.Core.Models.Api.Request.OrganizationSponsorships;
+using Bit.Core.OrganizationFeatures.OrganizationSponsorships.FamiliesForEnterprise.Interfaces;
+using Bit.Core.Repositories;
+using Microsoft.Extensions.Logging;
+using System;
+
+namespace Bit.Core.OrganizationFeatures.OrganizationSponsorships.FamiliesForEnterprise.Cloud
+{
+    public class CloudSyncOrganizationSponsorshipsCommand : CreateSponsorshipCommand, ICloudSyncOrganizationSponsorshipsCommand
+    {
+        private readonly IOrganizationSponsorshipRepository _organizationSponsorshipRepository;
+        private readonly IOrganizationUserRepository _organizationUserRepository;
+        private readonly ISendSponsorshipOfferCommand _sendSponsorshipOfferCommand;
+
+
+        public CloudSyncOrganizationSponsorshipsCommand (
+        IOrganizationSponsorshipRepository organizationSponsorshipRepository,
+        IOrganizationUserRepository organizationUserRepository,
+        ISendSponsorshipOfferCommand sendSponsorshipOfferCommand) : base(organizationSponsorshipRepository)
+        {
+            _organizationUserRepository = organizationUserRepository;
+            _organizationSponsorshipRepository = organizationSponsorshipRepository;
+            _sendSponsorshipOfferCommand = sendSponsorshipOfferCommand;
+        }
+
+        public async Task<OrganizationSponsorshipSyncModel> SyncOrganization(Organization sponsoringOrg, IEnumerable<OrganizationSponsorshipModel> sponsorshipsBatch)
+        {
+
+            var syncResponseModel = new OrganizationSponsorshipSyncModel
+            {
+                SponsoringOrganizationCloudId = sponsoringOrg.Id,
+            };
+
+
+            foreach (var selfHostedSponsorship in sponsorshipsBatch)
+            {
+                if (selfHostedSponsorship.SponsoringOrganizationUserId == null)
+                {
+                    throw new BadRequestException("Cannot sync sponsorship - missing user");
+                }
```

Comment:
Justin and I had a discussion about this, and since we are changing the `SponsoringOrganizationUserId` and `SponsoringOrganizationId` in the database to non nullable, we believe its acceptable to fail that entire sync batch if any of these are null.